tensorflow/lite/micro/kernels/shape.cc

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2017 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/kernels/op_macros.h"
+#include "tensorflow/lite/micro/kernels/kernel_util.h"
+#include "tensorflow/lite/micro/memory_helpers.h"
+#include "tensorflow/lite/micro/micro_utils.h"
+
+namespace tflite {
+namespace ops {
+namespace micro {
+namespace shape {
```

Comment:
TFLM is incrementally moving towards a flat tflite namespace (https://abseil.io/tips/130)

can you remove this hierarchy and put the file-local functions into an anonymous namespace?

See [full_connected.cc](https://github.com/tensorflow/tensorflow/pull/43682/files#diff-c3c2bd6c6974d231aec4cd63d94fe47b) from this PR as an initial template.

Reply:
Where should I place the prototype for Register_SHAPE()? Is the idea for every kernel to have a header file that will be included in micro_mutable_op_resolver.h?

Reply:
Good question. We'll move to a separate header on a case by csae basis, as needed for the code size optimization.

For this operator, let's keep Register_SHAPE if in  in micro_ops.h but move it out to the tflite namespace.

So micro_ops will become something like:

```cc

// Forward declaration of all micro op kernel registration methods. These
// registrations are included with the standard `BuiltinOpResolver`.
//
// This header is particularly useful in cases where only a subset of ops are
// needed. In such cases, the client can selectively add only the registrations
// their model requires, using a custom `(Micro)MutableOpResolver`. Selective
// registration in turn allows the linker to strip unused kernels.

namespace tflite {

// TFLM is incrementally moving towards a flat tflite namespace (https://abseil.io/tips/130). Any new ops (or cleanup of existing ops should have their Register function declarations in the tflite namespace.
TfLiteRegistration Register_SHAPE();


namespace ops {
namespace micro {
TfLiteRegistration Register_ABS();
TfLiteRegistration Register_ADD();
...

```

Reply:
Ok. All done.